app/src/main/java/com/nextcloud/talk/controllers/LocationPickerController.kt

```diff
@@ -214,19 +214,51 @@ class LocationPickerController(args: Bundle) :
     }
 
     private fun initMap() {
-        if (!isLocationPermissionsGranted()) {
-            requestLocationPermissions()
-        }
-
         binding.map.setTileSource(TileSourceFactory.MAPNIK)
-
         binding.map.onResume()
 
         locationManager = activity!!.getSystemService(Context.LOCATION_SERVICE) as LocationManager
-        try {
-            locationManager!!.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0L, 0f, this)
-        } catch (ex: SecurityException) {
-            Log.w(TAG, "Error requesting location updates", ex)
+
+        if (!isLocationPermissionsGranted()) {
+            requestLocationPermissions()
+        } else {
+            try {
+                when {
+                    locationManager!!.isProviderEnabled(LocationManager.NETWORK_PROVIDER) -> {
+                        locationManager!!.requestLocationUpdates(
+                            LocationManager.NETWORK_PROVIDER,
+                            MIN_LOCATION_UPDATE_TIME,
+                            MIN_LOCATION_UPDATE_DISTANCE,
+                            this
+                        )
+                    }
+                    locationManager!!.isProviderEnabled(LocationManager.GPS_PROVIDER) -> {
+                        locationManager!!.requestLocationUpdates(
+                            LocationManager.GPS_PROVIDER,
+                            MIN_LOCATION_UPDATE_TIME,
+                            MIN_LOCATION_UPDATE_DISTANCE,
+                            this
+                        )
+                        Log.d(TAG, "Using LocationManager.GPS_PROVIDER because LocationManager.NETWORK_PROVIDER was " +
+                            "not available")
+                    }
+                    else -> {
+                        Log.e(
+                            TAG,
+                            "Error requesting location updates. Probably this is a phone without google services and there" +
+                                " is no alternative like UnifiedNlp installed. Furthermore no GPS is supported."
+                        )
+                        Toast.makeText(context, context?.getString(R.string.nc_location_unknown), Toast.LENGTH_LONG)
```

Comment:
Fine for now since we make use of Toasts throughout the app. 

--> In the mid/long run we should move from `Toasts` to `Snackbars` which is the (actual) preferred way in Android.

Reply:
yes there should be one commit that replaces all toasts with snackbars 👍